src/Symfony/Component/Notifier/Bridge/Telegram/TelegramTransport.php

```diff
@@ -115,6 +126,14 @@ private function getPath(array $options): string
             isset($options['message_id']) => 'editMessageText',
             isset($options['callback_query_id']) => 'answerCallbackQuery',
             isset($options['photo']) => 'sendPhoto',
+            isset($options['location']) => 'sendLocation',
+            isset($options['audio']) => 'sendAudio',
+            isset($options['document']) => 'sendDocument',
```

Comment:
If I get this right, there is only one Endpoint for document, but we have `uploadDocument($filepath)` and `document($url)`, can you explain why we need both?

However, what I would check inside the methods without `upload` prefix, is, that the file exists. Please also add unit tests for this.

Reply:
Telegram supports 3 ways for sending files - http url, file_id and uploading via api request 

docs for photo for example - https://github.com/symfony/symfony/pull/51717/files#diff-d5c6f680237182e620c35f9de66750b1d9363cd6cd69032b68286945990fb4caR84

We can't detect is value file_id or file_path for upload. Before this pull request we had method `photo` which worked with http url and file_id, for avoiding BC we have to add uploadPhoto. For all other types i added same 2 methods to keep methods consistent for all types.

We can try to detect is argument `file_path` by checking `file_exists`, and if not pass value as `file_id`, but in this case if user will try to pass `file_path` to not existing file (because some application issues), he will not catch correct api error.

Reply:
Instead of using extra methods for uploading (`uploadPhoto`, `uploadDocument` and etc.) we can use flag `upload`:

```
  (new TelegramOptions())
    ->document('files/test.xlsx')
    ->upload();
```

In this case if `upload` flag exists then will handle argument as `filePath` if not as `fileId` or `httpUrl`

Reply:
I understand, from my POV we can go with the 2 methods, it's a clear API. 👍